app/test-utils/testFunctions/commonComponentsTests.js

```diff
@@ -0,0 +1,121 @@
+import { Selector, ClientFunction } from 'testcafe';
+import nock from 'nock';
+import dashboardWithCompleteSections from '../../../fixtures/dashboardWithCompleteSections.json';
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const titleTest = ({ pageSetup, sectionManifest }) => {
+  test('should render the page title', async (t) => {
+    await pageSetup({ t });
+    const title = Selector('[data-test-id="section-title"]');
+    await t
+      .expect(title.innerText).eql(sectionManifest.title);
+  });
+};
+
+const mainAdviceTest = ({ pageSetup, sectionManifest }) => {
+  test('should render main advice of section', async (t) => {
+    await pageSetup({ t });
+    const mainAdvice = Selector('[data-test-id="section-main-advice"]');
+    await t
+      .expect(mainAdvice.innerText).eql(sectionManifest.mainAdvice);
+  });
+};
+
+const allAdviceTest = ({ pageSetup, sectionManifest }) => {
+  test('should render all the advice of the section', async (t) => {
+    await pageSetup({ t });
+    const expectedAdditionalAdvice = sectionManifest.additionalAdvice.join('\n\n');
+    const additionalAdvice = Selector('[data-test-id="section-additional-advice"]');
+    await t
+      .expect(additionalAdvice.innerText).eql(expectedAdditionalAdvice);
+  });
+};
+
+const submitButtonTest = ({ pageSetup, sectionManifest }) => {
+  test('should render the submit button', async (t) => {
+    await pageSetup({ t });
+    const submitButton = Selector('[data-test-id="section-submit-button"]');
+    await t
+      .expect(submitButton.find('button').count).eql(1);
+    if (sectionManifest.submitText) {
```

Comment:
why are we guarding here?

Reply:
I thought some of them don't have submit text but I  just checked and they all do so I can remove the check